chain/chain/src/chain.rs

```diff
@@ -1060,6 +1066,100 @@ impl Chain {
         }
     }
 
+    /// Check if any orphan with missing chunks is ready to request chunks
+    pub fn check_orphans_with_missing_chunks<F>(
+        &mut self,
+        me: &Option<AccountId>,
+        mut orphans_missing_chunks: F,
+    ) where
+        F: Copy + FnMut((CryptoHash, EpochId, Vec<ShardChunkHeader>)) -> (),
+    {
+        let mut orphan_hashes = vec![];
+        for (orphan_hash, orphan) in
```

Comment:
Limit this one to `MAX_ORPHANS_REQUEST_CHUNKS` as well.

Reply:
I'm not sure which orphans the most important.
Then I think limiting here makes limiting below useless.
Let's discuss offline.